HRInvoiceApp/HRInvoiceApp/AddCompany.xaml.cs

```diff
@@ -19,27 +20,80 @@ public partial class AddCompany : ContentPage
 
         Department department;
         KvK kvk;
-        Province province;
+        List<Province> provinces;
         User user;
         
-        public AddCompany ()
+        public AddCompany()
 		{
-			InitializeComponent ();
+			InitializeComponent();
             SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Visible;
             db = App.Database.GetInstance();
 
             Task.Run(async () =>
             {
-                company = await db.Table<Company>().FirstOrDefaultAsync();
+                user = await db.Table<User>().FirstOrDefaultAsync();
+                provinces = await db.Table<Province>().ToListAsync();
+                if(provinces.Count == 0)
+                {
+                    // Error handling
+                }
+
+                if(user == null)
+                {
+                    await DisplayAlert("Alert", "Voeg eerst een gebruiker toe.", "OK");
+                    var modalPage = new SettingsPage();
+                    await Navigation.PushModalAsync(modalPage);
+                    return;
+                }
                 company = new Company();
+                kvk = new KvK();
+
+                ProvincePicker.ItemsSource = provinces;
+
             });
+
         }
 
         void saveCompany(object sender, EventArgs e)
         {
+            foreach (var view in addCompanyStackLayout.Children)
+            {
+                if (view is Entry)
```

Comment:
Geinige tip: Je kan ook `if (view is Entry entry)` doen ipv ` if (view is Entry)`. Dan hoef je op regel 63 niet meer ` Entry entry = (Entry)view;` te doen ;)